### DISCUSSION TITLE
getting current time?

### DISCUSSION THREAD

**Discussion by Tauum** [ndresx/react-countdown#174] [Q&A]
say if my timer is set to a specific time while a task us running.

But the task is completed before the timer is ended, how can i output the current time left remaining?

Im using this basic setup below:

```
const [timeLimit, setTimeLimit] = useState('');

useEffect(() => {
       setTimeLimit((30 * 1000) + Date.now()) // 1000 = seconds
    }
  }, [])

  const Completionist = () => <h4><span>Time has run out!</span></h4>;
  const renderer = ({ hours, minutes, seconds, completed }) => {
    if (completed) { return <Completionist />; }
    else { return <h4><span>{hours}:{minutes}:{seconds}</span></h4>; } // Render countdown
  };

  function finalize() {
    // do stuff
    })

return (
      <div className='quiz-main body'>

          <Countdown date={timeLimit} onComplete={finalize} className='timer'>
            <Completionist />
          </Countdown>

</div>
);

```

**Comment by Tauum** [comment]
Id also like to ask how we are supposed to call functions such as onPaused, how should this be referenced via JSX?

**Reply by ndresx** [reply]
I'm not sure what exactly you mean with "how can I output the current time left remaining" as this is what it should be doing. To answer your second question, you can find an example here of how to use it https://github.com/ndresx/react-countdown#api-reference.

**Reply by Tauum** [reply]
Sorry no, I mean output it to a variable when calling onPause, is that possible?

say the timer has 30 seconds, and onPause is called at 15 seconds, it should store 15 seconds was taken

**Reply by ndresx** [reply]
Yes, the `onPause` callback should receive the same object as for example the `renderer` callback. https://github.com/ndresx/react-countdown#onpause
